src/shared/pager.c

```diff
@@ -53,6 +53,11 @@ noreturn static void pager_fallback(void) {
         _exit(EXIT_SUCCESS);
 }
 
+static int stored_stdout = -1;
+static int stored_stderr = -1;
+static bool stdout_redirected = false;
+static bool stderr_redirected = false;
```

Comment:
If `close(STDOUT_FILENO)` is called before `pager_open` then `fcntl(STDOUT_FILENO, F_DUPFD_CLOEXEC, 3)` will return -1. So `stored_stdout` can be -1 even if `dup2(fd[1], STDOUT_FILENO)` doesn't fail.